server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/put/DecommissionRequest.java

```diff
@@ -21,25 +21,31 @@
 import static org.opensearch.action.ValidateActions.addValidationError;
 
 /**
- * Register decommission request.
- * <p>
  * Registers a decommission request with decommission attribute and timeout
  *
  * @opensearch.internal
  */
 public class DecommissionRequest extends ClusterManagerNodeRequest<DecommissionRequest> {
 
+    public static final TimeValue DEFAULT_REQUEST_TIMEOUT = TimeValue.timeValueMinutes(2L);
     public static final TimeValue DEFAULT_NODE_DRAINING_TIMEOUT = TimeValue.timeValueSeconds(120);
 
     private DecommissionAttribute decommissionAttribute;
-
+    private boolean retryOnClusterManagerSwitch = false;
+    private TimeValue timeout = DEFAULT_REQUEST_TIMEOUT;
     private TimeValue delayTimeout = DEFAULT_NODE_DRAINING_TIMEOUT;
 
     // holder for no_delay param. To avoid draining time timeout.
     private boolean noDelay = false;
 
     public DecommissionRequest() {}
 
+    public DecommissionRequest(DecommissionAttribute decommissionAttribute, boolean retryOnClusterManagerSwitch, TimeValue timeout) {
+        this.decommissionAttribute = decommissionAttribute;
+        this.retryOnClusterManagerSwitch = retryOnClusterManagerSwitch;
```

Comment:
nit:rename to `retryOnClusterManagerChange`

Reply:
Changed this based on this comment - https://github.com/opensearch-project/OpenSearch/pull/5542#discussion_r1057188313.